geoadmin/src/main/java/org/activityinfo/geoadmin/merge2/view/swing/match/select/CandidateTableModel.java

```diff
@@ -0,0 +1,44 @@
+package org.activityinfo.geoadmin.merge2.view.swing.match.select;
+
+import org.activityinfo.geoadmin.merge2.view.match.KeyFieldPairSet;
+import org.activityinfo.geoadmin.merge2.view.match.MatchGraph;
+import org.activityinfo.geoadmin.merge2.view.match.MatchSide;
+
+import javax.swing.table.AbstractTableModel;
+import java.util.List;
+
+class CandidateTableModel extends AbstractTableModel {
+    private final KeyFieldPairSet keyFields;
+    private final MatchSide side;
+    private final List<Integer> frontier;
+
+    public CandidateTableModel(MatchGraph graph, int fromIndex, MatchSide fromSide) {
+        this.keyFields = graph.getKeyFields();
+        this.side = fromSide;
+        frontier = graph.getParetoFrontier(fromIndex, fromSide);
+    }
+
+    public int getColumnCount() { return keyFields.size(); }
+
+    public int getRowCount() { 
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)